src/lib/isEAN.js

```diff
@@ -0,0 +1,70 @@
+/**
+ * The most commonly used EAN standard is
+ * the thirteen-digit EAN-13, while the
+ * less commonly used 8-digit EAN-8 barcode was
+ * introduced for use on small packages.
+ * EAN consists of:
+ * GS1 prefix, manufacturer code, product code and check digit
+ * Reference: https://en.wikipedia.org/wiki/International_Article_Number
+ */
+
+import assertString from './util/assertString';
+
+/**
+ * Define EAN Lenghts; 8 for EAN-8; 13 for EAN-13
+ * and Regular Expression for valid EANs (EAN-8, EAN-13),
+ * with exact numberic matching of 8 or 13 digits [0-9]
+ */
+const LENGTH_EAN_8 = 8;
+const validEanRegex = /(?<!\d)(\d{8}|\d{13})(?!\d)$/;
+
+
+/**
+ * Get position weight given:
+ * EAN length and digit index/position
+ *
+ * @param {number} length
+ * @param {number} index
+ * @return {number}
+ */
+function getPositionWeightThroughLengthAndIndex(length, index) {
+  if (length === LENGTH_EAN_8) {
+    return (index % 2 === 0) ? 3 : 1;
+  }
+
+  return (index % 2 === 0) ? 1 : 3;
+}
+
+/**
+ * Calculate EAN Check Digit
+ * Reference: https://en.wikipedia.org/wiki/International_Article_Number#Calculation_of_checksum_digit
+ *
+ * @param {string} ean
+ * @return {number}
+ */
+function calculateCheckDigit(ean) {
+  const checksum = ean
+    .slice(0, -1)
+    .split('')
+    .map((char, index) => Number(char) * getPositionWeightThroughLengthAndIndex(ean.length, index))
+    .reduce((acc, partialSum) => acc + partialSum, 0);
+
+  const remainder = 10 - (checksum % 10);
+
+  return remainder < 10 ? remainder : 0;
```

Comment:
The case where remainder equals 10 is uncovered in mocha tests. Can you add a test case?